File: sem2_FSharp/ViewModels/AuthtorizationModels/EmailModel.fs
namespace sem2_FSharp.ViewModels.AuthtorizationModels

open System.ComponentModel.DataAnnotations

type EmailModel() =
    [<DefaultValue>]
    val mutable private email: string

    [<Required(ErrorMessage = "Не указан Email")>]
    member public this.Email with get() = this.email
                                    and set p = this.email <- p

File: sem2/Models/ViewModels/AccountModels/EmailModel.cs
using System.ComponentModel.DataAnnotations;

namespace sem2.Models.ViewModels.AccountModels
{
    public class EmailModel
    {
        [Required(ErrorMessage = "Не указан email адрес")]
        [EmailAddress(ErrorMessage = "Не корректный email адрес")]
        [MaxLength(64)]
        public string Email { get; set; }
    }
}